Architecture/TodoApp/android/core/src/main/java/com/novoda/todoapp/tasks/data/model/Tasks.java

```diff
@@ -53,13 +53,21 @@ public boolean isEmpty() {
         return internalMap().isEmpty();
     }
 
-    public boolean hasNoActionMoreRecentThan(SyncedData<Task> syncedData) {
-        return hasNoActionMoreRecentThan(syncedData.data().id(), syncedData.lastSyncAction());
+    public boolean actionIsAbsentOrNoMoreRecentThan(SyncedData<Task> syncedData) {
+        return actionIsAbsentOrNoMoreRecentThan(syncedData.data().id(), syncedData.lastSyncAction());
     }
 
-    public boolean hasNoActionMoreRecentThan(Id id, long lastSyncAction) {
+    public boolean actionIsAbsentOrNoMoreRecentThan(Id id, long lastSyncAction) {
         SyncedData<Task> internalData = internalMap().get(id);
-        return internalData == null || lastSyncAction >= internalData.lastSyncAction();
+        return internalData == null || internalData.lastSyncAction() <= lastSyncAction;
+    }
+
+    public boolean actionExistsAndIsOutdatedComparedTo(SyncedData<Task> syncedData) {
```

Comment:
I agree we need to change those names, however we need to be able to show that the different is not just in the exists vs absent. They are also checking for lowerThan vs LowerThanOrEquals.